core_v2/sources/domains.move

```diff
@@ -156,7 +146,114 @@ module aptos_names::domains {
             register_name_events: account::new_event_handle<RegisterNameEventV1>(account),
         });
 
-        token_helper::initialize(account);
+        // Create collection + token_resource
+        let registry_seed = utf8_utils::u128_to_string((timestamp::now_microseconds() as u128));
+        string::append(&mut registry_seed, string::utf8(b"registry_seed"));
+        let (token_resource, token_signer_cap) = account::create_resource_account(
+            account,
+            *string::bytes(&registry_seed),
+        );
+        move_to(account, CollectionCapabilityV2 {
+            capability: token_signer_cap,
+        });
+        collection::create_unlimited_collection(
+            &token_resource,
+            utf8(COLLECTION_DESCRIPTION),
+            config::collection_name_v1(),
+            option::none(),
+            utf8(COLLECTION_URI),
+        );
+    }
+    
+    public fun get_token_signer_address(): address acquires CollectionCapabilityV2 {
+        account::get_signer_capability_address(&borrow_global<CollectionCapabilityV2>(@aptos_names).capability)
+    }
+
+    fun get_token_signer(): signer acquires CollectionCapabilityV2 {
+        account::create_signer_with_capability(&borrow_global<CollectionCapabilityV2>(@aptos_names).capability)
+    }
+
+    inline fun token_addr_inline(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): address acquires CollectionCapabilityV2 {
+        token::create_token_address(
+            &get_token_signer_address(),
+            &config::collection_name_v1(),
+            &token_helper::get_fully_qualified_domain_name(subdomain_name, domain_name),
+        )
+    }
+
+    public fun token_addr(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): address acquires CollectionCapabilityV2 {
+        token::create_token_address(
+            &get_token_signer_address(),
+            &config::collection_name_v1(),
+            &token_helper::get_fully_qualified_domain_name(subdomain_name, domain_name),
+        )
+    }
+
+    public fun get_record_obj(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): Object<NameRecordV2> acquires CollectionCapabilityV2 {
+        object::address_to_object(token_addr_inline(domain_name, subdomain_name))
+    }
+
+    inline fun get_record(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): &NameRecordV2 acquires CollectionCapabilityV2 {
+        borrow_global(token_addr_inline(domain_name, subdomain_name))
+    }
+
+    inline fun get_record_mut(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): &mut NameRecordV2 acquires CollectionCapabilityV2 {
+        borrow_global_mut(token_addr_inline(domain_name, subdomain_name))
+    }
+
+    /// Creates or force transfers the name token to `to_addr`
+    /// NOTE: This function does not do any validation
+    fun create_or_force_transfer_token(
```

Comment:
can we separate it to 2 functions? 1 handles not exist (create and transfer) 1 handles exist (transfer)